johnzon-mapper/src/main/java/org/apache/johnzon/mapper/ExceptionMessages.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.johnzon.mapper;
+
+import javax.json.JsonValue;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+import static javax.json.JsonValue.ValueType.ARRAY;
+import static javax.json.JsonValue.ValueType.FALSE;
+import static javax.json.JsonValue.ValueType.NUMBER;
+import static javax.json.JsonValue.ValueType.OBJECT;
+import static javax.json.JsonValue.ValueType.STRING;
+import static javax.json.JsonValue.ValueType.TRUE;
+
+public class ExceptionMessages {
+
+    private ExceptionMessages() {
+    }
+
+    public static String simpleName(final Type type) {
```

Comment:
Cant we use getTypeName instead to simplify it?

Reply:
I mentioned it a bit in JOHNZON-370 expecting feedback here.  My personal preference as a user is the short names of the classes involved as it greatly helps me "see" the exception text.  When the full packages are involved I am a slow reader so I start "scanning forward" and inevitably miss important exception text and it just takes me way longer and I find it overall more painful than helpful.

That said, the more detailed exception text that follows always does list the full package and class names and I intentionally left them long and didn't "fix" them.  I felt that was a good balance.  The first line is quick and short and filled with enough clues to be useful for normal cases.  The second line can help you in rare cases where a tiny bit more info is helpful.

Reply:
Simple name is ok for me and a substring, just having a simpler name extraction with more caseq handling is nicer from a code standpoint IMHO.